js/plugins/compat-oai/src/openai/index.ts

```diff
@@ -101,6 +205,59 @@ export const openAI = (options?: OpenAIPluginOptions) =>
         })
       );
     },
+    resolver,
+    listActions,
+  });
+}
+
+export type OpenAIPlugin = {
+  (params?: OpenAIPluginOptions): GenkitPlugin;
+  model(name: string, config?: any): ModelReference<z.ZodTypeAny>;
+  embedder(name: string, config?: any): EmbedderReference<z.ZodTypeAny>;
```

Comment:
Should this params be more tightly typed, like in googleai?

Reply:
probably a good idea because otherwise code completion for configs on various model types won't work well... feel free to do it as a follow up.